### PR TITLE
feat:  Integrate What's App

### LINKED ISSUE #740: [Feature Request] Integrate What's App
### Required prerequisites

- [X] I have searched the [Issue Tracker](https://github.com/camel-ai/camel/issues) and [Discussions](https://github.com/camel-ai/camel/discussions) that this hasn't already been reported. (+1 or comment there if it has.)
- [ ] Consider asking first in a [Discussion](https://github.com/camel-ai/camel/discussions/new).

### Motivation

different with working spaces like slack and discord, what's app is also popular used, we need to do some support for what's app

### Solution

_No response_

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Coder-Joe458**
## Description

Describe your changes in detail.

## Motivation and Context

 `close #740 ` 

- [ ] I have raised an issue to propose this change ([required](https://github.com/camel-ai/camel/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Implemented Tasks

- [ ] Subtask 1
- [ ] Subtask 2
- [ ] Subtask 3

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [x] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: camel/toolkits/whatsapp_toolkit.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+from typing import Any, Dict, List, Union
+
+import requests
+
+from camel.toolkits import OpenAIFunction
+from camel.toolkits.base import BaseToolkit
+
+
+class WhatsAppToolkit(BaseToolkit):
+    """A class representing a toolkit for WhatsApp operations.
```
Comment (Wendong-Fan) [inline camel/toolkits/whatsapp_toolkit.py:26]: docstring format, same as other parts
```suggestion
    r"""A class representing a toolkit for WhatsApp operations.
```

File: camel/toolkits/whatsapp_toolkit.py [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+        """
+        for attempt in range(self.retries):
+            try:
+                return func(*args, **kwargs)
+            except Exception as e:
+                print(f"Attempt {attempt + 1}/{self.retries} failed: {e}")
+                if attempt < self.retries - 1:
+                    time.sleep(self.delay)
+                else:
+                    raise
```
Comment (Wendong-Fan) [inline camel/toolkits/whatsapp_toolkit.py:86]: since this method can be generally used, we can move this method under `camel/utils/commons.py`

**Review by Wendong-Fan — COMMENTED** [review]
Thanks @Coder-Joe458 's contribution! Overall looks good, could you also add example code under `examples/function_call`? Thanks in advance!

File: camel/toolkits/whatsapp_toolkit.py (hunk 2 above)
Comment (Coder-Joe458) [reply camel/toolkits/whatsapp_toolkit.py]: Moved this to `camel/utils/commons.py`. Please review again. Thanks.

**Review by Wendong-Fan — APPROVED** [review]
Thanks @Coder-Joe458 ! Added one more commit to unify the docstring format here: https://github.com/camel-ai/camel/pull/972/commits/8edcd114f8c299b11596b8edab1c8ebe98e8feed

LGTM

**Pull request merged**